src/common/crypto/mbedtls/cryptoprovider.cpp

```diff
@@ -374,9 +374,103 @@ aos::Error MbedTLSCryptoProvider::ParseX509Certs(mbedtls_x509_crt* currentCrt, a
         return err;
     }
 
+    err = GetX509CertPublicKey(cert, currentCrt);
+    if (!err.IsNone()) {
+        return err;
+    }
+
     return GetX509CertExtensions(cert, currentCrt);
 }
 
+aos::Error MbedTLSCryptoProvider::GetX509CertPublicKey(aos::crypto::x509::Certificate& cert, mbedtls_x509_crt* crt)
+{
+    switch (mbedtls_pk_get_type(&crt->pk)) {
+    case MBEDTLS_PK_RSA:
+        return GetRSAKey(cert, mbedtls_pk_rsa(crt->pk));
+
+    case MBEDTLS_PK_ECKEY:
+        return GetECKey(cert, mbedtls_pk_ec(crt->pk));
+
+    default:
+        return aos::ErrorEnum::eNotFound;
+    }
+}
+
+aos::Error MbedTLSCryptoProvider::GetECKey(aos::crypto::x509::Certificate& cert, mbedtls_ecp_keypair* eckey)
```

Comment:
ditto for function name and param order.

Reply:
Done